src/shared/components/middle-header/middle-header.component.ts

```diff
@@ -12,12 +15,48 @@ import { Observable } from 'rxjs';
 export class MiddleHeaderComponent implements OnInit {
   amountInCart$: Observable<number>;
   amountOnWishlist$: Observable<number>;
+  user: IUser;
+  isDropdownOpened = false;
   protected readonly Route = Route;
 
-  constructor(private cartFacadeService: CartFacadeService, private productFacadeService: ProductFacadeService) {}
+  constructor(
+    private cartFacadeService: CartFacadeService,
+    private productFacadeService: ProductFacadeService,
+    private userFacadeService: UserFacadeService,
+    private cdr: ChangeDetectorRef,
+    private router: Router,
+    private destroyRef: DestroyRef,
+  ) {}
 
   ngOnInit(): void {
     this.amountInCart$ = this.cartFacadeService.getTotalAmountOfProductsInCart();
     this.amountOnWishlist$ = this.productFacadeService.getTotalAmountOnWishlist();
+
+    this.userFacadeService.isUserLoggedIn()
+      .pipe(takeUntilDestroyed(this.destroyRef))
+      .subscribe((user: IUser) => {
```

Comment:
isUserLoggedIn should return boolean value,
if it returns a user, we need to rename it